js/BoxMenuItemView.js

```diff
@@ -1,23 +1,34 @@
 import MenuItemView from 'core/js/views/menuItemView';
+import Adapt from 'core/js/adapt';
 import router from 'core/js/router';
 
 class BoxMenuItemView extends MenuItemView {
 
   className() {
-    return `${super.className()} boxmenu-item`;
+    return [
+      super.className(),
+      'boxmenu-item',
+      this.areEntireItemsClickable && 'is-entire-item-clickable'
+    ].filter(Boolean).join(' ');
   }
 
   events() {
     return {
-      'click .js-btn-click': 'onClickMenuItemButton'
+      'click .js-btn-click': 'onClickMenuItemButton',
+      'keydown .js-btn-click': 'onClickMenuItemButton'
     };
   }
 
   onClickMenuItemButton(event) {
     if (event && event.preventDefault) event.preventDefault();
+    if (event.code && (event.code !== 'Space' && event.code !== 'Enter')) return;
```

Comment:
```suggestion
    if (event.code && !['Space', 'Enter'].includes(event.code)) return;
    if (event?.preventDefault) event.preventDefault();
```
You probably want to prevent default only after the return early to stop other navigation keys from getting lost while the button has focus.

Reply:
@oliverfoster Yep, I think that fixed the issue that @kirsty-hames was having (the focus trap on the first item).

Reply:
> @oliverfoster Yep, I think that fixed the issue that @kirsty-hames was having (the focus trap on the first item).

I can confirm this is fixed thanks.